src/api/lock.ts

```diff
@@ -1,18 +1,54 @@
+import logger from 'pnpm-logger'
 import path = require('path')
-import thenify = require('thenify')
-import lockfile = require('lockfile')
-const lock = thenify(lockfile.lock)
-const unlock = thenify(lockfile.unlock)
+import lockfile = require('proper-lockfile')
 
-export default async function<T> (storePath: string, fn: () => Promise<T>): Promise<T> {
-  const lockfile: string = path.resolve(storePath, 'lock')
-  await lock(lockfile)
+async function delay(ms: number) {
+  return new Promise(resolve => setTimeout(resolve, ms))
+}
+
+async function lock(lockFilename: string, notFirstTime: boolean = false): Promise<{}> {
+  const promise = new Promise((resolve, reject) => {
+    lockfile.lock(
+      lockFilename,
+      {realpath: false},
+      async (err: Error & {code: string}) => {
+        if (err) {
+          if (err.code === 'ELOCKED') {
+            if (!notFirstTime) {
+              logger.warn('waiting for another installation to complete...')
```

Comment:
he/she should. Warnings are printed. Here is the related code https://github.com/pnpm/pnpm-reporter-default/blob/master/src/index.ts#L45-L48